packages/core/scripts/gas-rebate/index.js

```diff
@@ -385,11 +442,88 @@ async function getUmaPrice() {
+    const endTimeString = moment.unix(endTime).format("YYYY-MM-DD");
+
+    const query = `https://api.etherscan.io/api?module=stats&action=dailyavggasprice&startdate=${startTimeString}&enddate=${endTimeString}&sort=asc&apikey=${etherscanApiKey}`;
+    const response = await fetch(query, {
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json"
+      }
+    });
+
+    let data = (await response.json()).result;
+
+    // Return daily gas price (in Gwei) mapped to Unix timestamps so we can best estimate
+    // the gas price for each transaction.
+    const dailyPrices = data.map(_data => {
+      return { timestamp: Number(_data.unixTimeStamp), avgGwei: fromWei(_data.avgGasPrice_Wei, "gwei") };
+    });
+    return dailyPrices;
+  }
+}
+
+async function getHistoricalEthPrice(startBlock, endBlock) {
+  const etherscanApiKey = process.env.ETHERSCAN_API_KEY;
+  if (!etherscanApiKey) {
+    console.error("Missing ETHERSCAN_API_KEY in your environment, falling back to default ETH price");
+    return null;
+  } else {
+    const startTime = (await web3.eth.getBlock(startBlock)).timestamp;
+    const startTimeString = moment.unix(startTime).format("YYYY-MM-DD");
+    const endTime = (await web3.eth.getBlock(endBlock)).timestamp;
+    const endTimeString = moment.unix(endTime).format("YYYY-MM-DD");
+
+    const query = `https://api.etherscan.io/api?module=stats&action=ethdailyprice&startdate=${startTimeString}&enddate=${endTimeString}&sort=asc&apikey=${etherscanApiKey}`;
+    const response = await fetch(query, {
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json"
+      }
+    });
+
+    let data = (await response.json()).result;
+
+    // Return daily eth price mapped to Unix timestamps so we can best estimate
+    // the eth price for each transaction.
+    const dailyPrices = data.map(_data => {
+      return { timestamp: Number(_data.unixTimeStamp), avgPx: _data.value };
+    });
+    return dailyPrices;
+  }
+}
+
 // Implement async callback to enable the script to be run by truffle or node.
 async function Main(callback) {
   try {
+    const rebateNumber = 1;
+    const endDate = argv.end ? argv.end : Math.round(Date.now() / 1000 - 24 * 60 * 60); // Default: Current time minus 1 day.
+    const startDate = argv.start ? argv.start : endDate - 60 * 60 * 24 * 5; // Default: End time - 5 days
+    let endBlock, startBlock;
+    try {
+      endBlock = (await FindBlockAtTimestamp._findBlockNumberAtTimestamp(web3, Number(endDate))).blockNumber;
+      startBlock = (await FindBlockAtTimestamp._findBlockNumberAtTimestamp(web3, Number(startDate))).blockNumber;
+    } catch (err) {
+      throw err;
+    }
+
     // Pull the parameters from process arguments. Specifying them like this lets tests add its own.
-    await calculateRebate(argv.start, argv.end, argv["reveal-only"], argv["claim-only"]);
+    await calculateRebate({
```

Comment:
I think you should consider adding the `umaPrice`, `dailyAvgGasPrices`, `dailyAvgEthPrices` as parameters that can be fed in either from the `argv` or from a set of unit tests. This way the user can override these and you can also write tests.

Reply:
Done! Added to CI

Reply:
awesome, looks great!